pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -303,6 +303,20 @@ func ValidateFlowSchemaResourcePolicyRule(rule *flowcontrol.ResourcePolicyRule,
 		}
 	}
 
+	if len(rule.ResourceNames) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("resourceNames"), `resource names must supply at least one resource name`))
+	} else if hasWildcard(rule.ResourceNames) {
+		if len(rule.ResourceNames) > 1 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceNames"), rule.ResourceNames, "if '*' is present, must not specify other resource names"))
+		}
+	} else {
+		for idx, tgtName := range rule.ResourceNames {
+			if len(tgtName) == 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceNames").Index(idx), tgtName, "each member of this list must not be an empty string"))
```

Comment:
```suggestion
				allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceNames").Index(idx), tgtName, "must be non-empty"))
```

(since the field path already has index information)

Reply:
I think we can fix this and similar things in a followup (this is copy-paste code from the alpha type).

Reply:
Oh wait, this is the new field, not copy pasted. I think we should omit this in this PR and make a not-time-sensitive followup.

Reply:
will address this in https://github.com/kubernetes/kubernetes/pull/94244